### PR TITLE
Query identities associated with organization roles

### PR DESCRIPTION
**Pull request by tomfeigin**
Wrap queries to return the teams and users associated with a specific organization role

### CODE REVIEW

File: github/orgs_custom_roles.go [hunk 1]
```diff
@@ -126,3 +126,49 @@ func (s *OrganizationsService) DeleteCustomRepoRole(ctx context.Context, org, ro
 
 	return resp, nil
 }
+
+// ListTeamsAssignedToOrgRole returns all teams assigned to as specific organization role.
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:130]:
```suggestion
// ListTeamsAssignedToOrgRole returns all teams assigned to a specific organization role.
```

File: github/orgs_custom_roles.go [hunk 2]
```diff
@@ -126,3 +126,49 @@ func (s *OrganizationsService) DeleteCustomRepoRole(ctx context.Context, org, ro
...
+// In order to list teams assigned to an organization role, the authenticated user must be an organization owner.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/organization-roles#list-teams-that-are-assigned-to-an-organization-role
+//
+//meta:operation GET /orgs/{org}/organization-roles/{role_id}/teams
+func (s *OrganizationsService) ListTeamsAssignedToOrgRole(ctx context.Context, org string, roleID int64) ([]*Team, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/organization-roles/%v/teams", org, roleID)
+
+	req, err := s.client.NewRequest("GET", u, nil)
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:139]: According to this: https://docs.github.com/rest/orgs/organization-roles#list-teams-that-are-assigned-to-an-organization-role
this endpoint takes pagination options.

```suggestion
func (s *OrganizationsService) ListTeamsAssignedToOrgRole(ctx context.Context, org string, roleID int64, opts *ListOptions) ([]*Team, *Response, error) {
	u := fmt.Sprintf("orgs/%v/organization-roles/%v/teams", org, roleID)

	req, err := s.client.NewRequest("GET", u, opts)
```

File: github/orgs_custom_roles.go [hunk 3]
```diff
@@ -126,3 +126,49 @@ func (s *OrganizationsService) DeleteCustomRepoRole(ctx context.Context, org, ro
...
+	var teams []*Team
+	resp, err := s.client.Do(ctx, req, &teams)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return teams, resp, nil
+}
+
+// ListUsersAssignedToOrgRole returns all users assigned to as specific organization role.
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:153]:
```suggestion
// ListUsersAssignedToOrgRole returns all users assigned to a specific organization role.
```

File: github/orgs_custom_roles.go [hunk 4]
```diff
@@ -126,3 +126,49 @@ func (s *OrganizationsService) DeleteCustomRepoRole(ctx context.Context, org, ro
...
+// In order to list users assigned to an organization role, the authenticated user must be an organization owner.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/organization-roles#list-users-that-are-assigned-to-an-organization-role
+//
+//meta:operation GET /orgs/{org}/organization-roles/{role_id}/users
+func (s *OrganizationsService) ListUsersAssignedToOrgRole(ctx context.Context, org string, roleID int64) ([]*User, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/organization-roles/%v/users", org, roleID)
+
+	req, err := s.client.NewRequest("GET", u, nil)
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:162]: This one does too.

```suggestion
func (s *OrganizationsService) ListUsersAssignedToOrgRole(ctx context.Context, org string, roleID int64, opts *ListOptions) ([]*User, *Response, error) {
	u := fmt.Sprintf("orgs/%v/organization-roles/%v/users", org, roleID)

	req, err := s.client.NewRequest("GET", u, opts)
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @tomfeigin .
It looks like some changes need to be made, and the tests need to be updated to accommodate the pagination options.
No need to force-push, as we always squash & merge.

**Review by gmlewis — APPROVED** [review]
WHUPS!  Sorry my suggestions were bad... glad you caught them!!!
That will teach me to write a suggestion from memory!!!

Thank you, @tomfeigin !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

File: github/orgs_custom_roles.go [hunk 5]
```diff
@@ -126,3 +126,57 @@ func (s *OrganizationsService) DeleteCustomRepoRole(ctx context.Context, org, ro
...
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var teams []*Team
```
Comment (Sairaviteja27) [inline github/orgs_custom_roles.go:148]: I could find "notification_setting" in example response in API docs, but I could not find this field in Team struct.

Comment (tomfeigin) [reply github/orgs_custom_roles.go:148]: This is not relevant to this PR, you can open a new PR which adds fields to the `Team` struct

**Review by Sairaviteja27 — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @Sairaviteja27 !
Merging.

**Pull request merged**